KOGITO-7766 - Release notes for Data Index Addons (#4393)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/release-notes/ReleaseNotesKogito.1.33.0.Final/ReleaseNotesKogito.1.33.0.Final.adoc

// IMPORTANT: For 1.10 and later, save each version release notes as its own module file in the release-notes folder that this `ReleaseNotesKogito<version>.adoc` file is in, and then include each version release notes file in the chap-kogito-release-notes.adoc after Additional resources of {PRODUCT} deployment on {OPENSHIFT} section, in the following format:
//include::ReleaseNotesKogito.<version>/ReleaseNotesKogito.<version>.adoc[leveloffset=+1]

[id="ref-kogito-rn-new-features-1.33_{context}"]
= New features in {PRODUCT} 1.33

[role="_abstract"]
The following sections describe some of the new features or enhancements in {PRODUCT} 1.33.

== {PRODUCT} runtimes

=== {PRODUCT} Serverless Workflow

https://kiegroup.github.io/kogito-docs/serverlessworkflow/1.33.0.Final/release_notes.html[New features on 1.33].

== {PRODUCT} supporting services

=== Data Index Addons for {PRODUCT} runtimes

> Starting from this release, a new set of Quarkus extensions are available to incorporate Data Index features to {PRODUCT} runtimes. By allowing Data Index to be used as a Quarkus extension, any process and task execution will automatically be persisted using the Data Index extension storage in use. This requires that different storage types of the extension be provided. Below is the initial supported list of extensions:

Reviewer: ```suggestion
Starting from this release, a new set of Quarkus extensions are available to incorporate Data Index features to {PRODUCT} runtimes. By allowing Data Index to be used as a Quarkus extension, any process and task execution will automatically be persisted using the Data Index extension storage in use. This requires that different storage types of the extension be provided. The initial supported list of extensions is as follows:
```

> Starting from this release, a new set of Quarkus extensions are available to incorporate Data Index features to {PRODUCT} runtimes. By allowing Data Index to be used as a Quarkus extension, any process and task execution will automatically be persisted using the Data Index extension storage in use. This requires that different storage types of the extension to be provided. Below is the initial supported list of extensions:

Reviewer 2: ```suggestion
Starting from this release, a new set of Quarkus extensions are available to incorporate Data Index features to {PRODUCT} runtimes. By allowing Data Index to be used as a Quarkus extension, any process and task execution will automatically be persisted using the Data Index extension storage in use. This requires that different storage types of the extension be provided. Below is the initial supported list of extensions:
```

> Starting from this release, a new set of Quarkus extensions are available to incorporate Data Index features to {PRODUCT} runtimes. By allowing Data Index to be used as a Quarkus extension, any process and task execution will automatically be persisted using the Data Index extension storage in use. This requires that different storage types of the extension to be provided. Bellow is the initial supported list of extensions:

Reviewer 3: ```suggestion
Starting from this release, a new set of Quarkus extensions are available to incorporate Data Index features to {PRODUCT} runtimes. By allowing Data Index to be used as a Quarkus extension, any process and task execution will automatically be persisted using the Data Index extension storage in use. This requires that different storage types of the extension to be provided. Below is the initial supported list of extensions:
```

- kogito-addons-quarkus-data-index-infinispan
- kogito-addons-quarkus-data-index-mongodb
- kogito-addons-quarkus-data-index-inmemory
- kogito-addons-quarkus-data-index-postgresql

> To include any of these addons into your project, simply add the dependency as the example below, along with Quarkus required properties for the database connection to the application.properties.

Reviewer: ```suggestion
To include any of these addons into your project, just add the dependency as shown in the following example, along with Quarkus required properties for the database connection to the application properties.
```

.Data Index PostgreSQL Addon
[source,xml]
----
<dependencies>
  <dependency>
    <groupId>org.kie.kogito</groupId>
    <artifactId>kogito-addons-quarkus-data-index-postgresql</artifactId>
  </dependency>
</dependencies>
----

A new example has been included in our kogito-examples repository, to demonstrate how to combine a {PRODUCT} runtime with Data Index Addon using PostgreSQL persistence. For more details, visit https://github.com/kiegroup/kogito-examples/tree/stable/serverless-workflow-examples/serverless-workflow-data-index-quarkus[serverless-workflow-data-index-quarkus].

> By adding any of extensions to the runtime, indexing of the data starts to happen automatically, and the GraphQL endpoint becomes available to retrieve this data. In there, it will be possible to retrieve data from processes and tasks.

Reviewer 2: ```suggestion
By adding any extensions to the runtime, indexing of the data starts to happen automatically, and the GraphQL endpoint becomes available to retrieve this data. There, it will be possible to retrieve data from processes and tasks.
```

> Note: The Data Index Quarkus extension offers a subset of the schema available when compared to using the Data Index service. For instance, domain specific and jobs related queries are not available. Also, any GraphQL mutation target as Gateway API is also not supported.

Reviewer: ```suggestion
Note: The Data Index Quarkus extension offers a subset of the schema available when compared to using the Data Index service. For an instance, domain-specific and jobs-related queries are not available. Also, any GraphQL mutation target as Gateway API is also not supported.
```
